src/python/pants/backend/go/goals/test.py

```diff
@@ -540,6 +593,11 @@ async def run_go_tests(
             pkg_target_address=field_set.address,
         )
 
+    output_files = [x for x in output_files if x != "cover.out"]
+    if output_files:
+        # TODO: What is the best way to arrange for these files to be output?
```

Comment:
@benjyw / @Eric-Arellano: The remaining question in this PR is how exactly should ancillary profile files be written to disk?

Reply:
(given that the rule returns `TestResult`). The `extra_output` attribute does not appear to be used at all to write output out?

Reply:
hmm looks like it is written out https://github.com/pantsbuild/pants/blob/4dc2d025e8a7b2018e04a1e6ffde9a3f0b61cc80/src/python/pants/core/goals/test.py#L852-L855

Reply:
Latest commit just has the profiles be extra output from the test run.